src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -188,25 +188,21 @@ def _upgrade_on_windows():
     This also gives the user a chance to manually install the MSI in case of msiexec.exe failure.
     """
     import platform
+    import subprocess
+    import sys
 
     if platform.architecture()[0] == '32bit':
         msi_url = 'https://aka.ms/installazurecliwindows'
     else:
         msi_url = 'https://aka.ms/installazurecliwindowsx64'
     logger.warning("Updating Azure CLI with MSI from %s", msi_url)
-    tmp_dir, msi_path = _download_from_url(msi_url)
+    _, msi_path = _download_from_url(msi_url)
 
     logger.warning("Installing MSI")
-    import subprocess
-    exit_code = subprocess.call(['msiexec.exe', '/i', msi_path])
-
-    if exit_code:
-        logger.warning("Installation Failed. You may manually install %s", msi_path)
-    else:
-        from azure.cli.core.util import rmtree_with_retry
-        logger.warning("Succeeded. Deleting %s", tmp_dir)
-        rmtree_with_retry(tmp_dir)
```

Comment:
I suggest we find a way to preserve this logic, otherwise the temporary MSI will not be deleted.

Reply:
The MSI is in the `tempfile.mkdtemp()` directory (`C:\\Users\\xx\\AppData\\Local\\Temp\\tmp1s_cgvau`), it appears that windows can delete it automatically.
[Manage drive space with Storage Sense](https://support.microsoft.com/en-us/windows/manage-drive-space-with-storage-sense-654f6ada-7bfc-45e5-966b-e24aded96ad5)